packages/expect/src/utils.ts

```diff
@@ -268,16 +268,38 @@ export const subsetEquality = (
   object: any,
   subset: any,
 ): undefined | boolean => {
-  if (!isObjectWithKeys(subset)) {
-    return undefined;
-  }
+  const seenReferences = new WeakMap();
+
+  // subsetEquality needs to keep track of the references
+  // it has already visited to avoid infinite loops in case
+  // there are circular references in the subset passed to it.
+  const subsetEqualityWithContext = (
+    seenReferences: WeakMap<object, boolean>,
```

Comment:
How about moving the initialization of the weak map here?

Reply:
Good idea. Just pushed this change.

Btw the Azure builds seem to be very flaky on Windows machines due to their network connection 😢